src/Agent/NewRelic/Agent/Extensions/Providers/Wrapper/AwsSdk/DynamoDbRequestHandler.cs

```diff
@@ -43,8 +43,14 @@ private static string GetOperationNameFromRequestType(string requestType)
                 return _operationNameCache[requestType];
             }
             var operationName = requestType.Replace("Request", string.Empty).ToSnakeCase();
-            _operationNameCache.Add(requestType, operationName);
-            return operationName;
+            if (_operationNameCache.TryAdd(requestType, operationName))
```

Comment:
It might be simpler to just use GetOrAdd and replace most of the code in this method. Doing that would also avoid using the methods that are not necessarily thread safe.